aphrodite/engine/ray_tools.py

```diff
@@ -1,22 +1,46 @@
 """Ray for distributed multi-node inference: https://github.com/ray-project/ray"""
-import random
-from typing import List, Optional, Tuple
+import socket
+from typing import List, Optional, Tuple, TYPE_CHECKING
+
+from aphrodite.common.config import ParallelConfig
 
 try:
     import ray
+    from ray.air.util.torch_dist import TorchDistributedWorker
+    """Ray wrapper for aphrodite.task_handler.worker, allowing
+    worker to be lazily initialized after Ray sets CUDA_VISIBLE_DEVICES."""
+    class RayWorker(TorchDistributedWorker):
+        def __init__(self) -> None:
+            self.worker = None
+        
+        def init_worker(self, worker_init_fn):
+            self.worker = worker_init_fn()
+
+        def __getattr__(self, name):
+            return getattr(self.worker, name)
+        
+        def execute_method(self, method, *args, **kwargs):
+            executor = getattr(self, method)
+            return executor(*args, **kwargs)
+        
 except ImportError:
     ray = None
+    TorchDistributedWorker = None
 
-from aphrodite.common.config import ParallelConfig
-
-DeviceID = Tuple[int, Optional[str], int] # rank, node resource (node IP), device id
+if TYPE_CHECKING:
```

Comment:
Why is this only imported if a 3rd party type checker if checking the file?

Reply:
We want to avoid importing `PlacementGroup` during normal runtime, as it may not always be available and we may not always need it (i.e., we're not executing a multi-GPU worker). This should improve startup time and reduce memory overhead. 
Will set `RayWorker` to `None` so we can gracefully move on without crashing the script, give me a second.